File: week-09/rimrim990/모음사전.py
from bisect import bisect_right

N = 5
words = []

def solution(word):
  alphas = ['A', 'E', 'I', 'O', 'U']
  make_words('', 0, alphas)

  return bisect_right(words, word)

def make_words(word, idx, alphas):
  if len(word) == N:
    return

  for i in range(idx, len(alphas)):
    w = word + alphas[i]
    words.append(w)
    make_words(w, idx, alphas)

File: week-09/protoseo/모음 사전.java
import java.util.*;

class Solution {
    char[] vowel = {'A', 'E', 'I', 'O', 'U'};
    Set<String> words = new TreeSet<>();

    public int solution(String word) {
        find(0, "");
        List<String> wordsToList = new ArrayList<>(words);
        return wordsToList.indexOf(word);
    }

    public void find(int idx, String str) {
        words.add(str);
        if (idx == 5) {
            return;
        }
        for (int i = 0; i < 5; i++) {
            find(idx + 1, str + vowel[i]);
        }
    }
}
